### PR TITLE
[MIG] 11.0 -  partner auto salesman

### PR DESCRIPTION
**Pull request by JayVora-SerpentCS**
Migrated the Auto salesman module.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please respect commit history following migration guide.

**Comment by JayVora-SerpentCS** [comment]
@pedrobaeza Corrected it myself

**Comment by pedrobaeza** [comment]
OK, it seems so. Now tasks to do:

- Squash together adjacent "OCA Transbot..." commits
- Check Travis status.

**Comment by JayVora-SerpentCS** [comment]
Merged Transbot commits and travis goes green! Thanks @pedrobaeza

File: partner_auto_salesman/LICENSE/LICENSE [hunk 1]
```diff
@@ -0,0 +1,15 @@
+    OpenERP, Open Source Management Solution
```
Comment (yajo) [inline partner_auto_salesman/LICENSE/LICENSE]: I feel this file is not needed, and in any case this is not the AGPL license text.

File: partner_auto_salesman/README.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Partner Auto Salesman
+=====================
+
+This module adds the current logged in user to be the default salesman of the 
+partner being created.
+
+Installation
+============
+
+There are no special instructions regarding installation.
```
Comment (yajo) [inline partner_auto_salesman/README.rst]: Remove section then.

File: partner_auto_salesman/README.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+Configuration
+=============
+
+No configuration is needed.
```
Comment (yajo) [inline partner_auto_salesman/README.rst]: Remove too

File: partner_auto_salesman/README.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+* https://github.com/OCA/partner-contact/
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/134/11.0
+
+Known issues / Roadmap
+======================
+
+* No known issues.
```
Comment (yajo) [inline partner_auto_salesman/README.rst]: 🔥

File: partner_auto_salesman/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
```
Comment (yajo) [inline partner_auto_salesman/__init__.py]: Use [headers from template](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3) please

File: partner_auto_salesman/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
```
Comment (yajo) [inline partner_auto_salesman/__manifest__.py]: Same

File: partner_auto_salesman/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+{
+    'name': 'Partner auto salesman - v11',
+    'version': '11.0.0.1.0',
+    'category': 'Customer Relationship Management',
+    'complexity': "easy",
+    'description': """Fill the salesman field with the current user id""",
```
Comment (yajo) [inline partner_auto_salesman/__manifest__.py]: Remove, we have a README

File: partner_auto_salesman/models/__init__.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
```
Comment (yajo) [inline partner_auto_salesman/models/__init__.py]: Same as above

File: partner_auto_salesman/models/res_partner.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
```
Comment (yajo) [inline partner_auto_salesman/models/res_partner.py]: same

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by JayVora-SerpentCS** [comment]
Ping @Yajo

File: partner_auto_salesman/models/res_partner.py [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo import fields, models
+
+
+class Partner(models.Model):
+    _inherit = 'res.partner'
+
+    user_id = fields.Many2one('res.users',
+                              string='Salesperson',
+                              help='The internal user that is in charge of'
```
Comment (Yajo) [inline partner_auto_salesman/models/res_partner.py]: Finish with a space, or we'll get `ofcommunicating`

File: partner_auto_salesman/models/res_partner.py [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class Partner(models.Model):
+    _inherit = 'res.partner'
+
+    user_id = fields.Many2one('res.users',
```
Comment (yajo) [inline partner_auto_salesman/models/res_partner.py:10]: This field already exists upstream, so don't redefine everything; just the things that change (actually, the default value).

**Review by yajo — CHANGES_REQUESTED** [review]
In general terms, this addon is not strictly needed.

You can set default values in a per user basis through the debug menu in i.e. the partners view:

captura de pantalla de 2018-02-08 08-36-23
captura de pantalla de 2018-02-08 08-37-07

But I guess it can have sense under some circumstances.

In any case, there's still something remaining.

File: partner_auto_salesman/__manifest__.py [hunk 12]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Partner auto salesman - v11',
+    'version': '11.0.0.1.0',
+    'category': 'Customer Relationship Management',
+    'complexity': "easy",
+    'author': "Savoir-faire Linux,Odoo Community Association (OCA)",
+    'website': 'http://www.savoirfairelinux.com',
```
Comment (nikul-serpentcs) [inline partner_auto_salesman/__manifest__.py:10]: IMHO Please change URL as described in https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10

File: partner_auto_salesman/__init__.py [hunk 13]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (NL66278) [inline partner_auto_salesman/__init__.py:1]: From 11.0 onwards (python 3.x) no more coding lines needed.

**Review by NL66278 — CHANGES_REQUESTED** [review]
See my comment and those of others.

**Comment by NL66278** [comment]
No action on comments.

**Pull request closed without merging**